src/lib/support/CommonPersistentData.h

```diff
@@ -73,17 +74,40 @@ struct StoredDataList : public PersistentData<kMaxSerializedSize>
     }
 };
 
-constexpr size_t kPersistentFabricBufferMax = 32;
-struct FabricList : StoredDataList<FabricIndex, kPersistentFabricBufferMax>
+constexpr size_t kPersistentFabricListBufferMax = 32;
+struct FabricList : StoredDataList<FabricIndex, kPersistentFabricListBufferMax>
 {
-    // Subclasses need to define UpdateKey to be whatever fabric list key they
-    // care about.
+    CHIP_ERROR UpdateKey(chip::StorageKeyName & key) override
+    {
+        key = DefaultStorageKeyAllocator::GroupFabricList();
+        return CHIP_NO_ERROR;
+    }
 
     void Clear() override
     {
         first_entry = kUndefinedFabricIndex;
         entry_count = 0;
     }
+
+    CHIP_ERROR UnregisterFabric(PersistentStorageDelegate * mStorage, FabricIndex fabric_index, FabricIndex next)
+    {
```

Comment:
Definitely not. This method relies on an external caller to provide both the removing fabric, and the next fabric, with no guarantee of consistency.